### DISCUSSION TITLE
"Purge more" popup when changing / loading filament

### DISCUSSION THREAD

**Discussion by Schnabulation** [Jyers/Marlin#136] [Ideas]
Hello all

The normal Marlin LCD as a popup asking to purge more filament after a filament change and before resuming printing.
I find this a very interesting feature as the preconfigured purge length of advance pause in configuration_adv.h may sometimes not be enough (ex. switching from darker to lighter filament).

The behaviour is as follow:
1. Loading or changing filament
2. Slow load then fast load is performed
3. Filament is purged according to `ADVANCED_PAUSE_PURGE_LENGTH` in configuration_adv.h
4. Popup asking to purge more filament, if pressed "yes", more filament is purged (I don't know how much).
5. Popup is asking once again to purge more filament, if pressed "yes", more filament is purged and loop is started again.
6. Once pressed "no", printer continues to print.

I would love to see something like this implemented.

**Comment by t3chguy** [comment]
1-3 are from the depths of M600 - 4-6 could possibly be handled by this rewrite, not sure how they'd block the print from continuing before M600 "completes" though

**Reply by t3chguy** [reply]
It might be a case of "this isn't something Marlin Extensible UI supports yet"

**Comment by Jyers** [comment]
I found the code for this buried in M600 as @t3chguy noted. It would seem this is restricted to full MarlinUI systems. I have been looking at perhaps doing a second migration of the firmware to get it fully implemented into MarlinUI, but this will depend on how much time I have. Look out for a new branch if you are interested in any development I do towards a full MarlinUI approach.

**Reply by Schnabulation** [reply]
Thank you, I appreciate it...

I‘m not too familiar with the Marlin code but I assume there is no way to make such a feature work for our display, is it?

**Reply by Jyers** [reply]
Either the display code needs to move over to Marlin UI, or the M600 code needs to be modified. I would prefer option one, but that probably won't be in the immediate future.

**Reply by Schnabulation** [reply]
I see, that‘s unfortunate but understandable.

Can be closed / deleted / eaten / send to moon / whatever :-D
